internal/olm/operator/bundle/install.go

```diff
@@ -200,93 +195,82 @@ func (i *Install) setup(ctx context.Context) error {
 }
 
 // addBundleToIndexImage adds the bundle to an existing index image if the bundle is not already present in the index image.
-func addBundleToIndexImage(indexImage, bundleImage string) (*declarativeconfig.DeclarativeConfig, error) {
-	var bundleDeclConfig *declarativeconfig.DeclarativeConfig
+func addBundleToIndexImage(ctx context.Context, indexImage string, bundleDeclConfig *declarativeconfig.DeclarativeConfig) (*declarativeconfig.DeclarativeConfig, error) {
 	render := action.Render{
 		Refs: []string{indexImage},
 	}
 
 	log.Infof("Rendering a File-Based Catalog of the Index Image")
 
-	imageDeclConfig, err := render.Run(context.TODO())
+	imageDeclConfig, err := render.Run(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	// render the bundle image to a declarative config.
-	render = action.Render{
-		Refs: []string{bundleImage},
-	}
-
-	bundleDeclConfig, err = render.Run(context.TODO())
-	if err != nil {
-		log.Errorf("error in rendering the bundle image: %v", err)
-		return nil, err
-	}
-
 	if len(bundleDeclConfig.Bundles) < 0 {
 		log.Errorf("error in rendering the correct number of bundles: %v", err)
 		return nil, err
 	}
 
-	// check if the package blob already exists in the image
-	packageNotPresent := true
-	if len(bundleDeclConfig.Packages) > 0 {
-		for _, packageName := range imageDeclConfig.Packages {
-			if reflect.DeepEqual(packageName, bundleDeclConfig.Packages[0]) {
-				packageNotPresent = false
-				break
+	// check if the bundle exists in the index image.
+	bundleNotPresent := true
+	if len(bundleDeclConfig.Channels) > 0 && len(bundleDeclConfig.Bundles) > 0 {
+		for _, channel := range imageDeclConfig.Channels {
+			// Find the specific channel that the bundle needs to be inserted into
+			if channel.Name == bundleDeclConfig.Channels[0].Name && channel.Package == bundleDeclConfig.Channels[0].Package {
+				// Check if the CSV name is already present in the channel's entries
+				for _, entry := range channel.Entries {
+					if entry.Name == bundleDeclConfig.Bundles[0].Name {
+						bundleNotPresent = false
+						return nil, fmt.Errorf("Bundle image %s already present in the Index Image: %s", bundleDeclConfig.Bundles[0].Name, indexImage)
```

Comment:
A couple of points/questions here
1. If we're choosing to error out, then we probably can do away with the flag completely. This means we would also remove it from the condition in the code below. 
2. As we discussed, I reverted my implementation back to simply serving the index image. There are some reasons for this that I'd like to discuss.